nodejs/awsx/apigateway/README.md

```diff
@@ -0,0 +1,139 @@
+
+To define an endpoint you will either need to specify the routes or a swagger string. You can also define the stage name (else it will default to "stage"). A `stage` is an addressable instance of the Rest API.
+
+### Routes
+
+The destination is determined by the route, which can be an Event Handler Route, a Static Route, an Integration Route or a Raw Data Route.
+
+#### Event Handler Route
+
+An Event Handler Route is a route that will map to a [Lambda](https://aws.amazon.com/lambda/). You will need to specify the path, method and the lambda. Pulumi allows you to define the lambda inline with your Pulumi code & provisions that appropriate permissions.
+
+```ts
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+let endpoint = new awsx.apigateway.API("example", {
+    routes: [{
+        path: "/",
+        method: "GET",
+        eventHandler: async (event) => {
+            return {
+                statusCode: 200,
+                body: "hello",
+            };
+        },
+    }],
+})
+```
+
+A complete example can be found [here](https://github.com/pulumi/pulumi-awsx/blob/master/nodejs/awsx/examples/api/index.ts).
+
+#### Static Route
+
+A Static Route is a route that will map to static content in files/directories. You will need to define the local path & then the files (and subdirectories) will be uploaded into S3 objects. If the local path points to a file, you can specify the content-type. Else, the content types for all files in a directory are inferred.
+
+By default, any request on directory will serve index.html. This behavior can be disabled, by setting the index field to false.
+
+```ts
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+let endpoint = new awsx.apigateway.API("example", {
+    routes: [{
+        path: "/www",
+        localPath: "www",
+        index: false,
+    }],
+})
+```
+
+A complete example can be found [here](https://github.com/pulumi/pulumi-awsx/blob/master/nodejs/awsx/examples/api/index.ts).
+
+#### Integration Route
+
+[TODO]
+
+#### Raw Data Route
+
+[TODO]
+
+### Request Validation
+
+API Gateway can perform basic validations against request parameters, a request payload or both. When a validation fails, a 400 error is returned immediately.
+
+#### Validators
+
+Validators can be assigned at the API level or at the method level. The validators defined at a method level override any validator set at the API level.
+
+You must specify one of the following validators:
+
+* "ALL" - validate both the request body and request parameters
+* "BODY_ONLY" - validate only the request body
+* "PARAMS_ONLY" - validate only the request parameters
+
+```ts
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+let endpoint = new awsx.apigateway.API("example", {
+    routes: [{
+        path: "/www",
+        localPath: "www",
+        index: false,
+        requestValidator: "PARAMS_ONLY",
+    }],
+    requestValidator: "ALL",
```

Comment:
i'm not sure i understand the example since you have the requestValidator both at the route level and the API level.  Does this mean that the route option overrides the API option?  Or vice versa?

Reply:
Route validators override the API validator. Maybe I can add comments to the examples.
`{
        path: "/www",
        localPath: "www",
        requiredParameters: [{
            name: "key",
            in: "query",
        }],
        /**
         * These parameters will not actually be required since the method level validator is
         * set to only validate the body.
         */
        requestValidator: "BODY_ONLY", // This will override the API level requestValidator.
    }`

Reply:
Comments would be great :)